### ISSUE TITLE
API frontend authorization doesn't take parameters into consideration

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#3475]
The API frontend authorization layer performs hash lookups on paths in order to figure out the admin role associated to the path but when this path contains a parameter, then the hash lookup doesn't work.

Instead, when this hash lookup fails, it should attempt to check if it starts with any of the defined paths in the ACL map.

I'll also validate the performance since it will certainly be slower than only the hash lookup

**Comment by julsemaan** [comment]
Benchmark for the current devel for a path that is known:
```
BenchmarkIsAuthorizedAdminRolesStatic-8            10000            198511 ns/op
```

Same for a path that isn't known (with a parameter at the end)
```
BenchmarkIsAuthorizedAdminRolesDynamic-8            5000            379590 ns/op
```

**Comment by julsemaan** [comment]
Should have the fix check for the prefix and a "/" to prevent issue outlined here
https://github.com/inverse-inc/packetfence/commit/6f70b08b7a53a320f5df0c6471686f7f07b56bb9#diff-f0a4e8b18b23968ff18c74855a5a13aeR218

**Comment by julsemaan** [comment]
fixed via e73016eea5c85423fcdfb41d2eb5e2520dd7ffb7 and 0c284927bdc433795ca59fd2ea949ba3e40a0f9d
